test/feature_library/test_feature_library.py

```diff
@@ -225,7 +232,9 @@ def test_sindypi_library_bad_params(params):
         pytest.lazy_fixture("data_custom_library_bias"),
         pytest.lazy_fixture("data_generalized_library"),
         pytest.lazy_fixture("data_ode_library"),
-        pytest.lazy_fixture("data_pde_library"),
+        # pytest.lazy_fixture("data_pde_library"),
+        # we should not use this library for lorenz data,
+        # since spatiotemporal grid determines derivative order
```

Comment:
"order" was not the right word to use here, but the issue is that the PDELibrary inspects the shape of the spatiotemporal grid (`self.ind_range = spatiotemporal_grid.ndim - 1`) to determine which derivatives to include in the library. This old test incorrectly supplied a library that used a 1d spatiotemporal grid to use on the Lorenz ODE data (a 0d system).  The old test passed before we changed some reshaping, but it would have produced a nonsensical fit (the data was forced into the right shape in a nonsensical way). When we changed the reshaping procedure, it began to throw an error, which just pointed out that the test didn't make sense in the first place. I could have just deleted the pytest.lazy_fixture and not included comments, but that may leave some case untested. A new test_fit_transform_pde could also be made to use pde data to test fit_transform with the pde and weakpdelibrary.

Reply:
PS: this points out that fit should really check the shapes of the data to ensure it is compatibile with the spatial/spatiotemporal grid, and raise an interpretable error if it isn't. Eventually I would like to allow different spatiotemporal grids to each trajectory in the multiple_trajectory case, which will probably require the spatial grid to be passed not at the initialization but later in the fit. But I'm planning to open a new pull request for this in the future.

Reply:
Sounds good - so should we add remove the comment, & add an issue to move stgrid initialization to `fit()`?  We don't need to solve this during the PR, I just want to make sure everything noticed during the PR ends up tracked.

Reply:
This sounds good to me. I think the feature_library.fit can't take additional arguments other than x and y, so what makes sense to me is that model.fit includes optional grid options, which, if present, calls a new feature_library.set_grid function to set the grid. Then, when feature_library.fit is called, it checks to see if the grid is None (then, use a default grid) or if it is not a compatible shape as the input data (then, raise an error).

Reply:
I added an issues #222 for the future. Feel free to push a change removing the test comments, and we can resolve this conversation. I will continue reviewing in the meanwhile.

Reply:
done :heavy_check_mark: